File: backend/models/weekly_prompt.go
package models

import "gorm.io/gorm"

// WeeklyPrompt holds the single active prompt for a given ISO week.
// Only one row per iso_week (enforced by uniqueIndex).
type WeeklyPrompt struct {
	gorm.Model
	ISOWeek    string `gorm:"uniqueIndex" json:"iso_week"`
	PromptText string `json:"prompt_text"`
	SetByID    uint   `json:"set_by_id"`
	SetByAlias string `json:"set_by_alias"`
	SetByRole  string `json:"set_by_role"`
}

// PromptResponse is one user reply in the current week's discussion thread.
type PromptResponse struct {
	gorm.Model
	PromptID  uint   `json:"prompt_id"`
	UserID    uint   `json:"user_id"`
	UserAlias string `json:"user_alias"`
	UserRole  string `json:"user_role"`
	Content   string `json:"content"`
}

File: frontend/lib/models/weekly_prompt.dart
class WeeklyPrompt {
  final int id;
  final String isoWeek;
  final String promptText;
  final int setById;
  final String setByAlias;
  final String setByRole;
  final DateTime createdAt;

  WeeklyPrompt({
    required this.id,
    required this.isoWeek,
    required this.promptText,
    required this.setById,
    required this.setByAlias,
    required this.setByRole,
    required this.createdAt,
  });

  factory WeeklyPrompt.fromJson(Map<String, dynamic> json) {
    return WeeklyPrompt(
      id: json['ID'] as int,
      isoWeek: json['iso_week'] as String,
      promptText: json['prompt_text'] as String,
      setById: (json['set_by_id'] as int?) ?? 0,
      setByAlias: (json['set_by_alias'] as String?) ?? '',
      setByRole: (json['set_by_role'] as String?) ?? '',
      createdAt: DateTime.parse(json['CreatedAt'] as String),
    );
  }
}
